ogr/ogrsf_frmts/geojson/ogrgeojsonreader.cpp

```diff
@@ -3157,3 +3189,171 @@ json_object *CPL_json_object_object_get(struct json_object *obj,
     json_object_object_get_ex(obj, key, &poRet);
     return poRet;
 }
+
+bool OGRGeoJSONGetExtent(json_object *poObj, OGREnvelope *poEnvelope)
```

Comment:
It seems to me that having just OGRGeoJSONGetExtent3D() should be sufficient, with slight modification to handle the 2D & 3D cases. For example by declaring dfX, dfY, dfZ variables, and doing the Merge() outside the loop to be independent of the number of ordinates in the array